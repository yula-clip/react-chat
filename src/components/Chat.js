import React from "react";
import { withRouter } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import ChatMessageList from "./ChatMessageList";
import MessageInput from "./MessageInput";

const styles = theme => ({
  chatLayout: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    paddingTop: "64px",
    height: "100%",
    width: "100%",
    overflow: "hidden"
  }
});

const Chat = ({
  classes,
  messages,
  activeChat,
  activeUser,
  joinChat,
  sendMessage,
  isConnected
}) => (
  <main className={classes.chatLayout}>
    <ChatMessageList messages={messages} activeUser={activeUser} />
    {activeChat && (
      <MessageInput
        disabled={!isConnected}
        sendMessage={sendMessage}
        showJoinButton={!activeUser.isChatMember}
        onJoinButtonClick={() => joinChat(activeChat._id)}
        activeUser={activeUser}
      />
    )}
  </main>
);

export default withRouter(withStyles(styles)(Chat));
