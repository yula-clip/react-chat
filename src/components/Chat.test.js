/* eslint-env jest */
import React from "react";
import { MemoryRouter } from "react-router-dom";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Chat from "./Chat";

jest.mock("./ChatMessageList", () => () => "ChatMessageList");
jest.mock("./MessageInput", () => () => "MessageInput");

const mockProps = {
  activeUser: {
    firstName: "Name",
    lastName: "Surname",
    username: "username",
    isMember: true,
    isCreator: true,
    isChatMember: true
  },
  activeChat: {
    _id: "12345",
    title: "My Chat"
  },
  messages: [
    {
      chatId: "12345",
      content: "content",
      sender: {},
      createdAt: "2018-03-16T10:53:23.200Z"
    }
  ],
  joinChat: jest.fn(),
  isConnected: true,
  sendMessage: jest.fn()
};

describe("<Chat />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Chat {...mockProps} />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly ", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Chat {...mockProps} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
