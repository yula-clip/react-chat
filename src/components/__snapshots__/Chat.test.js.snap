// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly  1`] = `
<main
  className="Chat-chatLayout-2"
>
  ChatMessageList
  MessageInput
</main>
`;
